packages/forms/test/form_control_spec.ts

```diff
@@ -312,6 +312,56 @@ describe('FormControl', () => {
       c.setValidators([Validators.required]);
       expect(c.validator).not.toBe(null);
     });
+
+    it('should check presence of and remove a validator set in the control constructor', () => {
+      const c = new FormControl('', Validators.required);
+      expect(c.hasValidator(Validators.required)).toEqual(true);
+
+      c.removeValidators(Validators.required);
+      expect(c.hasValidator(Validators.required)).toEqual(false);
+
+      c.addValidators(Validators.required);
+      expect(c.hasValidator(Validators.required)).toEqual(true);
+    });
+
+    it('should check presence of and remove a validator set with addValidators', () => {
+      const c = new FormControl('');
```

Comment:
nit: add a check that there is no validator present initially:
```suggestion
      const c = new FormControl('');
      expect(c.hasValidator(Validators.required)).toEqual(false);
      
```